common/appointment/cancel.ts

```diff
@@ -0,0 +1,44 @@
+import { lecture as Appointment, lecture_appointmenttype_enum as AppointmentType } from '@prisma/client';
+import { prisma } from '../prisma';
+import { getStudent, User } from '../user';
+import * as Notification from '../notification';
+import { getLogger } from '../logger/logger';
+import { getAppointmentForNotification } from './util';
+
+const logger = getLogger('Appointment');
+
+export async function cancelAppointment(user: User, appointment: Appointment) {
+    await prisma.lecture.update({
+        data: { isCanceled: true },
+        where: { id: appointment.id },
+    });
+
+    logger.info(`Appointment(${appointment.id}) was cancelled by User(${user.userID})`);
+
+    const student = await getStudent(user);
+
+    switch (appointment.appointmentType) {
```

Comment:
replaced if...else by switch to later turn on exhaustiveness checking for switch on union types